openwisp_monitoring/monitoring/migrations/__init__.py

```diff
@@ -26,15 +28,22 @@ def assign_permissions_to_groups(apps, schema_editor):
     for model_name in operators_read_only_admins_manage:
         try:
             permission = Permission.objects.get(codename='view_{}'.format(model_name))
+            if model_name in ['check', 'alertsettings']:
+                permission = Permission.objects.get(
+                    codename='view_{}_inline'.format(model_name)
+                )
             operator.permissions.add(permission.pk)
         except Permission.DoesNotExist:
             pass
         for operation in manage_operations:
-            admin.permissions.add(
-                Permission.objects.get(
-                    codename='{}_{}'.format(operation, model_name)
-                ).pk
+            permission = Permission.objects.get(
+                codename='{}_{}'.format(operation, model_name)
             )
+            if model_name in ['check', 'alertsettings']:
+                permission = Permission.objects.get(
+                    codename='{}_{}_inline'.format(operation, model_name)
+                )
+            admin.permissions.add(permission.pk)
```

Comment:
I think it would be simpler and clearer to add new data migration code instead of editing the existing data migrations.
Moving around logic which deals with data migrations is error prone and not worth the effort at the moment.

Reply:
Ok, I'll create a separate data migration in `monitoring` app named `add_inline_permissions.py` :+1: